src/components/instance-form-builder/GeneralInfoForm.jsx

```diff
@@ -180,9 +180,13 @@ class GeneralInfoForm extends React.Component {
             }
             const fieldKeys = fields.map(field => field.name);
             const errorMessages = await getValidationMessages(d2, instance, fieldKeys);
+            const connectionErrors = await instance.check();
```

Comment:
Do you think this should be all part of the validations flow or separate like it's here?

Reply:
I guess this should be moved to save() and handle this kind of validations there.

We could find potential performance issues if the user does loads of changes and not being able to cancel running Promises.

Reply:
There are some valid alternatives. As Alexis says, it makes sense to move the logic the model. I'd rename it to `checkAndSave()` (calls check + save), and keep a `save()` without the check.

Another functional option is to add a separate button that just tests the instance and leave the save without a check. Use case: you want to create an instance record, but the instance is not up at the moment for some reason. Check this with @adrianq, though.